pageserver/src/tenant/timeline.rs

```diff
@@ -3608,10 +3267,17 @@ impl Timeline {
                     );
                 }
             }
-            let mut layer_paths: Vec<PathBuf> = new_layers.iter().map(|l| l.path()).collect();
+
+            // FIXME: the writer already fsyncs all data, only rename needs to be fsynced here
+            let mut layer_paths: Vec<PathBuf> = new_layers
+                .iter()
+                .map(|l| l.local_path().to_owned())
+                .collect();
 
             // Fsync all the layer files and directory using multiple threads to
             // minimize latency.
+            //
+            // FIXME: spawn_blocking above for this
```

Comment:
I guess the earlier FIXME in line 3039 is about this one here?

Reply:
Yes I was doing a bit of back and forth with if we should really have the sapwn_blocking or not, how to efficiently read the values to it if reading would would become async. Could remove this...